pulsar/consumer_multitopic.go

```diff
@@ -149,8 +149,10 @@ func (c *multiTopicConsumer) ReconsumeLater(msg Message, delay time.Duration) {
 	fqdnTopic := internal.TopicNameWithoutPartitionPart(names[0])
 	consumer, ok := c.consumers[fqdnTopic]
 	if !ok {
-		c.log.Warnf("consumer of topic %s not exist unexpectedly", msg.Topic())
-		return
+		if consumer, ok = c.consumers[names[0].Name]; !ok {
```

Comment:
Let's make a variable for names[0]
```
tn := names[0]
```
Also, let's add a comment.
```
// check to see if the topic with the partition part is in the consumers
// this can happen when the consumer is configured to consume from a specific partition
```

Reply:
done